src/start_server.js

```diff
@@ -13,24 +13,54 @@ var http = require('http');
 var makeApp = require('./make_app');
 var open = require('open');
 var util = require('util');
+var findPort = require('find-port');
 
-function startServer(port, openPage, componentDir, packageName) {
-  port = port || 8080;
-  console.log('Starting Polyserve on port ' + port);
+function startWithPort(options) {
+
+  options.port = options.port || 8080;
+
+  console.log('Starting Polyserve on port ' + options.port);
 
   var app = express();
-  var polyserve = makeApp(componentDir, packageName);
+  var polyserve = makeApp(options.componentDir, options.packageName);
 
   app.use('/components/', polyserve);
 
   var server = http.createServer(app);
-  server = app.listen(port);
 
-  var baseUrl = util.format('http://localhost:%d/components/%s/', port, polyserve.packageName);
+  server = app.listen(options.port);
+
+  server.on('error', function(err) {
+    if (err.code === 'EADDRINUSE')
+      console.error("ERROR: Port in use", options.port, "Aborting!");
+    process.exit(69);
+  });
+
+  var baseUrl = util.format('http://localhost:%d/components/%s/', options.port,
+    polyserve.packageName);
   console.log('Files in this directory are available under ' + baseUrl);
 
-  if (openPage) {
-    open(baseUrl + openPage);
+  if (options.page) {
+    var url = baseUrl + (options.page === true ? 'index.html' : options.page);
+    if (Array.isArray(options.browser)) {
+      for (var i=0; i<options.browser.length; i++)
```

Comment:
that's the way I like it too, but g-w-c team has been beating me over the head about it. done,